pyartcd/pyartcd/pipelines/promote.py

```diff
@@ -92,12 +94,7 @@ async def run(self):
                 raise ValueError("No arches specified.")
             # Get previous list
             upgrades_str: Optional[str] = group_config.get("upgrades")
-            if upgrades_str is None and assembly_type != assembly.AssemblyTypes.CUSTOM:
-                raise ValueError(f"Group config for assembly {self.assembly} is missing the required `upgrades` field. If no upgrade edges are expected, please explicitly set the `upgrade` field to empty string.")
-            previous_list = list(map(lambda s: s.strip(), upgrades_str.split(","))) if upgrades_str else []
-            # Ensure all versions in previous list are valid semvers.
-            if any(map(lambda version: not VersionInfo.isvalid(version), previous_list)):
-                raise ValueError("Previous list (`upgrades` field in group config) has an invalid semver.")
+            previous_list = self.validate_upgrades(upgrades_str, assembly_type, self._doozer_working_dir / "ocp-build-data")
```

Comment:
Would be nice to catch the exceptions that can happen so the slack messsage does not contain a backtrace, but actionable info.

The other verifications do things with `justifications`, any reason this patch is deviating from that pattern?

Reply:
Fair points. No justification for deviation, I just left it out of the first draft :) I'll get that in